brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java

```diff
@@ -381,21 +394,37 @@ public boolean isDerivedFrom(String nodeId, Alien4CloudApplication toscaApplicat
         return getTemplatePropertyObjects(relationshipTemplate, paasNodeTemplate, builtPaaSNodeTemplates, toscaApplication.getKeywordMap(nodeTemplate, relationshipTemplate));
     }
 
-    private String resolveAttribute(Map.Entry<String, IValue> attribute, Alien4CloudApplication toscaApplication, PaaSNodeTemplate paaSNodeTemplate, Map<String, PaaSNodeTemplate> allNodes) {
-        String returnValue = FunctionEvaluator.parseAttribute(
-                attribute.getKey(),
-                attribute.getValue(),
-                toscaApplication.getTopology(),
-                ImmutableMap.<String, Map<String, InstanceInformation>>of(),
-                "",
-                paaSNodeTemplate,
-                allNodes);
-        // Alien4Cloud returns the attribute key surrounded by <> if it does not evaluate to anything
-        // We turn this to the empty String to avoid populating sensors.
-        if (returnValue.equals("<" + attribute.getKey() + ">")) {
-            return "";
+    private Object resolveAttribute(Map.Entry<String, IValue> attribute, Alien4CloudApplication toscaApplication, String nodeId, PaaSNodeTemplate paaSNodeTemplate, Map<String, PaaSNodeTemplate> allNodes) {
+        Map<String, PaaSNodeTemplate> builtPaaSNodeTemplates = getAllNodes(toscaApplication);
+        PaaSNodeTemplate paasNodeTemplate = builtPaaSNodeTemplates.get(nodeId);
+        IValue attributeValue = attribute.getValue();
+        if (attributeValue instanceof AttributeDefinition) {
+            String defaultValue = ((AttributeDefinition) attribute.getValue()).getDefault();
+            if (Strings.isBlank(defaultValue)) {
+                return null;
```

Comment:
I mean that A4C UI (or the plugin, I'm not sure which) will set each of the TOSCA attributes defined on an entity to the empty String.  If we try to create a static sensor for each of those attributes they conflict with the existing sensors, but all that happens is that the value is set on the existing sensor, which is a problem if the existing sensor is not a String sensor.